js/playlist.js

```diff
@@ -1,9 +1,11 @@
 define(['ytDataAPI', 'ytIframeAPI', 'template','jquery'],
 function(ytData,ytIframe,template,$) {
     var playlist=[];
+    var durations=[];
+    var repeat = [];
```

Comment:
I don't feel that the repeat option should be specific to every playlist. Because it is more of a choice on how the user wants to listen to the current playlist in general. I can't think of a case when a user may want to loop for a certain playlist, and not loop or shuffle for a different one.